syntax = "proto3";

package dfs;

service ChunkService {
  // Unary RPC for single chunk
  rpc StoreChunk(Chunk) returns (StoreResponse) {};
  
  // Client-side streaming for batches
  rpc StoreChunkStream(stream Chunk) returns (StoreResponse) {};

  rpc GetChunk (ChunkRequest) returns (ChunkResponse);
}

message Chunk {
  string id = 1;
  bytes data = 2;
}

message StoreResponse {
  bool success = 1;
  string message = 2;
}

message ChunkRequest {
  string chunk_uuid = 1;
}

message ChunkResponse {
  bytes data = 1;
  bool success = 2;
}